src/main/java/com/salesforce/storm/spout/dynamic/DynamicSpout.java

```diff
@@ -185,18 +202,19 @@ public void open(Map topologyConfig, TopologyContext topologyContext, SpoutOutpu
     @Override
     public void nextTuple() {
         // Report any errors
-        getCoordinator()
+        getMessageBus()
             .getErrors()
             .ifPresent(throwable -> getOutputCollector().reportError(throwable));
 
-        // Ask the SpoutCoordinator for the next message that should be emitted. If it returns null, then there's
+        // Ask the MessageBus for the next message that should be emitted. If it returns null, then there's
         // nothing new to emit! If a Message object is returned, it contains the appropriately mapped MessageId and
         // Values for the tuple that should be emitted.
-        final Message message = getCoordinator().nextMessage();
-        if (message == null) {
+        final Optional<Message> messageOptional = getMessageBus().nextMessage();
```

Comment:
Optionals feel unnecessary here, I would rather we chuck check for null in this case.